agent.go

```diff
@@ -501,17 +508,21 @@ func (a *Agent) watchHealthChecks(nodeListCh chan map[string]bool) {
 }
 
 func (a *Agent) getHealthChecks(waitIndex uint64, nodes map[string]bool) (api.HealthChecks, uint64) {
-	namespaces, err := namespacesList(a.client)
+	namespaces, err := namespacesList(a.client, a.config)
 	if err != nil {
 		a.logger.Warn("Error getting namespaces, falling back to default namespace", "error", err)
 		namespaces = []*api.Namespace{{Name: ""}}
 	}
 
 	ctx, cancelFunc := context.WithCancel(context.Background())
-	opts := (&api.QueryOptions{
+	opts := &api.QueryOptions{
 		NodeMeta:  a.config.NodeMeta,
 		WaitIndex: waitIndex,
-	}).WithContext(ctx)
+	}
+	opts = opts.WithContext(ctx)
+	a.HasPartition(func(partition string) {
```

Comment:
so following up this could be re-written as:
```
opts.Partition = a.PartitionOrEmtpy()
```
and removes the need for passing callback functions around and has the same effect

Reply:
Initially, I implemented that way but OSS/CE version fails to comprehend event empty partition, hence throwing errors about the partitions are only supported on Enterprise edition only. That make me change the code to this instead of using many if conditions.